### ISSUE TITLE
Rule proposal: no-extraneous-slashes

### ISSUE THREAD

**Issue by danny-andrews** [benmosher/eslint-plugin-import#915]
It would be nice to have a rule which catches extraneous slashes in paths. It could either be one rule: `no-extraneous-slashes` with options `allowMultiple` and `allowTrailing` (both defaulting to false) or two: `no-multiple-slashes`/`no-trailing-slashes`.

### Fail
```js
import blah from './src//util';
import React from 'react/';
```

Related to https://github.com/benmosher/eslint-plugin-import/issues/557.

I'm leaning towards one rule.

**Comment by ljharb** [comment]
"Multiple" would be handled by #912 (ie, ensuring the shortest path).

"Trailing" isn't safe, because `'./foo'` and `'./foo/'` might be very different modules, depending on if `foo/index.js` coexists with `foo.js` (and any other require.extensions) - so to implement that, the rule would have to check for any cases where the two things didn't resolve to the same path based on what's on disk.

However, #912's solution would presumably handle that as well, so I'm going to close this as a duplicate of #471.

**Comment by danny-andrews** [comment]
Sick! I found that PR first, but wasn't sure if it covered these cases.

**Comment by ljharb** [comment]
If it doesn't, it should :-) i believe there is some existing discussion on it related to "trailing".

**Comment by graingert** [comment]
It covers the // case. I've not fixed it to do the trailing slash...  Yet

**Comment by wottpal** [comment]
> It covers the // case. I've not fixed it to do the trailing slash... Yet

Should be probably reopened then?

**Comment by ljharb** [comment]
@wottpal since that comment 6 years ago, the PR was adapted to include that, so no. If you have a problem with the latest version, please file a new issue.
